terraform/modules/cloudfoundry/elb_uaa.tf

```diff
@@ -94,41 +94,46 @@ resource "aws_wafv2_web_acl" "cf_uaa_waf_core" {
   }
 
   # New rule for dropping logging for a specific host
-
-  rule {
-    name     = var.waf_label_host_0
-    priority = 0
-
-    action {
-      allow {}
-    }
-
-    statement {
-      byte_match_statement {
-        field_to_match {
-          single_header {
-            name = "host"
+  dynamic "rule" {
+    for_each = var.waf_hostnames_0
+    iterator = app_name
+    content {
+      name     = var.waf_label_host_0
+      priority = 0
+      action {
+        allow {}
+      }
+      visibility_config {
+        cloudwatch_metrics_enabled = "true"
+        metric_name                = var.waf_label_host_0
+        sampled_requests_enabled   = "true"
+      }
+      statement {
+        or_statement {
+          dynamic "statement" {
+            for_each = var.waf_hostnames_0
```

Comment:
Wait, is the nested `for_each` intentional here? With this code, I'd expect we'd get a separate rule for each hostname and three separate statements (combined with OR) within each rule

If the goal is to make the rules totally optional based on whether there are values in the variable, then `rule` needs to be dynamic. If the goal is just to have a dynamic set of statements within a rule that is always deployment, then `statement` here is the only thing that needs to be dynamic